src/database/utilities/numberSequence.js

```diff
@@ -0,0 +1,71 @@
+import { NUMBER_SEQUENCE_KEYS } from './constants';
+import { sortDataBy } from '../../utilities';
+
+const clearNumberSequences = database => {
+  database.write(() => {
+    const existingNumberSequences = database.objects('NumberSequence');
+    database.delete('NumberSequence', existingNumberSequences);
+    const existingNumberToReuse = database.objects('NumberToReuse');
+    database.delete('NumberToReuse', existingNumberToReuse);
+  });
+};
+
+const getMaxSerialNumber = (database, sequenceKey) => {
+  switch (sequenceKey) {
+    case NUMBER_SEQUENCE_KEYS.SUPPLIER_INVOICE_NUMBER: {
+      const transactions = database.objects('Transaction');
+      const filteredData = transactions.filtered(
+        '((type == $0 OR type == $1) AND otherParty.type != $2)',
+        'supplier_invoice',
+        'supplier_credit',
+        'inventory_adjustment'
+      );
+      const sortedData = sortDataBy(filteredData.slice(), 'serialNumber', false);
+      return sortedData.length <= 0 ? 0 : sortedData[0].serialNumber;
+    }
+    case NUMBER_SEQUENCE_KEYS.STOCKTAKE_SERIAL_NUMBER: {
+      const stocktake = database.objects('Stocktake');
+      const sortedData = sortDataBy(stocktake.slice(), 'serialNumber', false);
+      return sortedData.length <= 0 ? 0 : sortedData[0].serialNumber;
+    }
+    case NUMBER_SEQUENCE_KEYS.REQUISITION_REQUESTER_REFERENCE: {
```

Comment:
The serial number is used to increment both request and response requisitions. This request reference is a kind of serial number for request requisitions only. The idea is to add a reference number to request requestions so that it would become requisition 1...2... and so on. When a response is created for this request requisition, this requester_reference serial number is set to there_ref of the response requisition. This would link the request and response requisition.